### PR TITLE
refactor context setters, and document them properly

### PR DESCRIPTION
**Pull request by beniwohli**
This also adds functionality to copy transaction context data to errors that happen during the transaction.

### CODE REVIEW

File: docs/api.asciidoc [hunk 1]
```diff
@@ -185,8 +185,49 @@ import elasticapm
 elasticapm.set_transaction_data({'billing_amount': product.price * item_count})
```
Comment (simitt) [inline docs/api.asciidoc]: Is it really still `set_transaction_data` and not `set_context`?

File: tests/client/client_tests.py [hunk 2]
```diff
@@ -650,3 +650,18 @@ def test_transaction_max_span_nested(should_collect, elasticapm_client):
...
+    elasticapm.set_custom_context({'a': 'b'})
+    elasticapm.set_user_context(username='foo', email='foo@example.com', user_id=42)
+    elasticapm_client.capture_message('x', custom={'foo': 'bar'})
+    transaction = elasticapm_client.end_transaction('test', 'OK')
+    message = elasticapm_client.events[0]['errors'][0]
+    assert message['context']['custom'] == {'a': 'b', 'foo': 'bar'}
+    assert message['context']['user'] == {'username': 'foo', 'email': 'foo@example.com', 'id': 42}
+    assert message['context']['tags'] == {'foo': 'baz'}
+    assert 'a' in transaction.context['custom']
+    assert 'foo' not in transaction.context['custom']
```
Comment (simitt) [inline tests/client/client_tests.py]: Maybe I am missing it but I don't see a test that checks that you can call the update functions multiple times. as described in the api docs.

File: docs/api.asciidoc (hunk 1 above)
Comment (beniwohli) [reply docs/api.asciidoc]: great catch!

File: tests/client/client_tests.py (hunk 2 above)
Comment (beniwohli) [reply tests/client/client_tests.py]: Excellent point, I added some tests :)

**Pull request closed without merging**